plugin/Categorifier/Core.hs

```diff
@@ -77,6 +77,13 @@ import System.IO.Unsafe (unsafePerformIO)
 conversionFunction :: TH.Name
 conversionFunction = 'Categorifier.Categorify.expression
 
+conversionFunctionArity :: Int
+#if MIN_VERSION_GLASGOW_HASKELL(9, 0, 0, 0)
+conversionFunctionArity = 6
```

Comment:
Could you explain why this difference was needed?

Reply:
The arity of the function the rule is being applied to needs to be provided when you install the plugin. This is the arity in the Core representation, so that means it includes type and kind parameters, too. In GHC 9+, the conversion function (`Categorifier.Categorify.expression`) is multiplicity-polymorphic (i.e., linear or not), so there is an extra parameter for that multiplicity in GHC 9+.

Reply:
Got it. Thanks for the explanation.